lib/elixir/lib/gen_server.ex

```diff
@@ -159,6 +159,54 @@ defmodule GenServer do
   the GenServer callbacks as doing so will cause the GenServer to misbehave.
   If you want to receive custom messages, always receive them in `handle_info/2`.
 
+  ## Debugging with `:sys` module
+
+  GenServers, as all [special processes](http://erlang.org/doc/design_principles/spec_proc.html)
+  can be debugged using the `:sys` module. Through various hooks, it allows
+  developers to introspect into the inner state of the process and
+  trace system events that happen during it's execution, such as received
+  messages or sent replies.
```

Comment:
Could we mention state changes when tracing as that is perhaps the most useful to know.